pilot/pkg/model/status/helper.go

```diff
@@ -101,21 +97,16 @@ func DeleteConfigCondition(cfg config.Config, condition string) config.Config {
 	}
 	cfg = cfg.DeepCopy()
 	status := cfg.Status.(*v1alpha1.IstioStatus)
-	status.Conditions = DeleteCondition(status.Conditions, condition)
+	status.Conditions = deleteCondition(status.Conditions, condition)
 	return cfg
 }
 
-func DeleteCondition(conditions []*v1alpha1.IstioCondition, condition string) []*v1alpha1.IstioCondition {
+func deleteCondition(conditions []*v1alpha1.IstioCondition, condition string) []*v1alpha1.IstioCondition {
 	ret := append([]*v1alpha1.IstioCondition(nil), conditions...)
-	idx := -1
 	for i, cond := range ret {
 		if cond.Type == condition {
-			idx = i
-			break
+			ret = slices.Delete(conditions, i)
```

Comment:
I think we need to break after this, otherwise the `i` will be wrong if there are multiple matching conditions?

Or if we want to handle bothwe can use slices.FilterInPlace? Or avoid the clone and just do slices.Filter?

Reply:
Yes. I think we can just do slices.Filter